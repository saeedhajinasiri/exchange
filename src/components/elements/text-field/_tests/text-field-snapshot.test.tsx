import React from "react";
import {cleanup, render} from "test";
import TextField from "../text-field";

afterEach(cleanup)

describe('Test TextField UI', () => {
  it('Default', () => {
    const {asFragment} = render(<TextField/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with className prop', () => {
    const {asFragment} = render(<TextField className={"test-class-name"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with placeholder prop', () => {
    const {asFragment} = render(<TextField placeholder={"test-placeholder"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with label prop', () => {
    const {asFragment} = render(<TextField label={"test-label"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TextField Colors', () => {
  it('color=primary', () => {
    const {asFragment} = render(<TextField color={"primary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=secondary', () => {
    const {asFragment} = render(<TextField color={"secondary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=tertiary', () => {
    const {asFragment} = render(<TextField color={"tertiary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=grey', () => {
    const {asFragment} = render(<TextField color={"grey"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TextField Input Modes', () => {
  it('type=text', () => {
    const {asFragment} = render(<TextField type={"text"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('type=number', () => {
    const {asFragment} = render(<TextField type={"number"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TextField Variants', () => {
  it('variants="outlined', () => {
    const {asFragment} = render(<TextField variant={"outlined"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('variants="standard', () => {
    const {asFragment} = render(<TextField variant={"standard"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TextField Disabled', () => {
  it('disabled=true', () => {
    const {asFragment} = render(<TextField disabled={true}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('disabled=false', () => {
    const {asFragment} = render(<TextField disabled={false}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TextField Error', () => {
  it('error=true', () => {
    const {asFragment} = render(<TextField error={true}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('error=false', () => {
    const {asFragment} = render(<TextField error={false}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('error=true,errorMessage=sometext', () => {
    const {asFragment} = render(<TextField error={false} errorMessage={'some text'}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
