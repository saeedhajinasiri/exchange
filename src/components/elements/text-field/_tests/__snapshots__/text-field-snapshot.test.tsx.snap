// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextField Colors color=grey 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-grey border-grey bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Colors color=primary 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Colors color=secondary 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-secondary border-secondary bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Colors color=tertiary 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-tertiary border-tertiary bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Disabled disabled=false 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Disabled disabled=true 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 border-grey bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      disabled=""
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Error error=false 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Error error=true 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 border-tertiary bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Error error=true,errorMessage=sometext 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Input Modes type=number 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="number"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Input Modes type=text 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField UI Default 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField UI Test with className prop 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] test-class-name"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField UI Test with label prop 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      placeholder=" "
      type="text"
      value=""
    />
    <label
      class="flex z-[1] w-full select-none absolute text-grey
          leading-tight peer-focus:leading-tight peer-placeholder-shown:leading-[4]
          peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-danger 
          peer-disabled:before:border-transparent peer-disabled:after:border-transparent
          transition-all top-0 peer-focus:translate-y-[-50%] text-m-sm before:w-3
          before:translate-y-[50%] after:translate-y-[50%]
          before:mr-1 after:ml-1
          peer-placeholder-shown:before:border-transparent peer-placeholder-shown:after:border-transparent
            !before:peer-focus:border-t-0 !after:peer-focus:border-t-0  before:peer-focus:border-primary after:peer-focus:border-primary"
    >
      test-label
    </label>
  </div>
</DocumentFragment>
`;

exports[`Test TextField UI Test with placeholder prop 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      placeholder="test-placeholder"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Variants variants="outlined 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border hover:bg-transparent focus:border-2 rounded"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Test TextField Variants variants="standard 1`] = `
<DocumentFragment>
  <div
    class="relative min-w-[200px] undefined"
  >
    <input
      class="relative z-[2] peer w-full h-full transition-all px-3 py-2.5 focus:border-t-transparent h-10 sm:h-12 bg-primary border-primary  bg-transparent border-x-none border-t-none !focus:border-x-none !border-b !rounded-none"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
